modyn/selector/internal/selector_strategies/abstract_downsample_strategy.py

```diff
@@ -23,11 +24,44 @@ class AbstractDownsampleStrategy(AbstractPresampleStrategy):
     def __init__(self, config: dict, modyn_config: dict, pipeline_id: int, maximum_keys_in_memory: int):
         super().__init__(config, modyn_config, pipeline_id, maximum_keys_in_memory)
 
-        if "downsampled_batch_size" not in self._config:
-            raise ValueError("To use downsampling strategies, you have to specify the downsampled_batch_size")
-        self.downsampled_batch_size = self._config["downsampled_batch_size"]
+        if "sample_then_batch" not in config:
+            raise ValueError(
+                "Please specify if you want to sample and then batch or vice versa. "
+                "Use the sample_then_batch parameter"
+            )
+        if config["sample_then_batch"]:
+            self.downsampling_mode = DownsamplingMode.SAMPLE_THEN_BATCH
+        else:
+            self.downsampling_mode = DownsamplingMode.BATCH_THEN_SAMPLE
 
-        if not isinstance(self.downsampled_batch_size, int):
-            raise ValueError("The downsampled batch size must be an integer")
+        if self.downsampling_mode == DownsamplingMode.BATCH_THEN_SAMPLE and "downsampling_period" in config:
+            raise ValueError("downsampling_period can be used only in sample-then-batch.")
+
+        self.downsampling_period = config.get("downsampling_period", 1)
+
+        if "downsampled_batch_ratio" not in config:
+            raise ValueError("Please specify downsampled_batch_ratio to use downsampling methods")
+
+        self.downsampled_batch_ratio = config["downsampled_batch_ratio"]
+
+        if not (0 < self.downsampled_batch_ratio < 100) or not isinstance(self.downsampled_batch_ratio, int):
+            raise ValueError("The downsampled batch ratio must be an integer in (0,100)")
 
         self._requires_remote_computation = True
+
+    def get_downsampling_scale(self) -> float:
+        if self.downsampling_mode == DownsamplingMode.SAMPLE_THEN_BATCH:
+            return self.downsampled_batch_ratio / 100
+        return 1
```

Comment:
OK, this function is not the clearest, but it is not needed for training. It only serves to update the status bar. We only need this parameter for STB since in the real training loop we only see get_downsampling_scale()*total_datapoints points (and thus the status bar adjusts accordingly). In BTS, in the real training loop we see all datapoints so this parameter is not needed. I rename the function in a more sensible way indicating that it affects the status bar.

Reply:
I have changed the name of this function to `get_status_bar_scale` so as not to cause confusion and I have also added a comment in the function